### ISSUE TITLE
Neovim freezes while saving and working with large files in splits.

### ISSUE THREAD

**Issue by Preetam-Das** [neoclide/coc.nvim#3672]
## Result from CocInfo

```
## versions

vim version: NVIM v0.7.0-dev+1136-g005a7aa16
node version: v17.6.0
coc.nvim version: 0.0.80-3fb4c1f976
coc.nvim directory: /home/pree/.local/share/nvim/plugged/coc.nvim
term: alacritty
platform: linux

## Log of coc.nvim

2022-02-27T00:00:14.805 INFO (pid:6778) [services] - registered service "pyright"
2022-02-27T00:00:14.817 INFO (pid:6778) [services] - Pyright Server state change: stopped => starting
2022-02-27T00:00:14.819 INFO (pid:6778) [plugin] - coc.nvim initialized with node: v17.6.0 after 169ms
2022-02-27T00:00:14.976 INFO (pid:6778) [services] - Pyright Server state change: starting => running
2022-02-27T00:00:14.988 INFO (pid:6778) [services] - service pyright started
2022-02-27T00:00:17.601 INFO (pid:6778) [completion-complete] - 1 results from: pyright
2022-02-27T00:00:17.683 INFO (pid:6778) [completion-complete] - 1 results from: pyright
2022-02-27T00:01:54.820 INFO (pid:6778) [completion-complete] - 0 results from:
2022-02-27T00:02:22.698 INFO (pid:6778) [attach] - receive notification: showInfo []

```
## Describe the bug

Every time I work with a split which has large no. of words in it and try to save the file neovim gets stuck on ':w' and takes about 3 - 5 to save. To reproduce the bug i have to edit the file which has coc running.

This issue may seem like a duplicate of #1015 but I have the latest node installed. Although I am using neovim-nightly, but i tried the stable release and also tried downgrading node but nothing seems to work. 

While neovim is stuck like that, node seems to use 100% of CPU rescource as shown in the htop screenshot below. 
I am not sure how to go about solving this issue please help ...

## Reproduce the bug

The following mini.vim is enough to reproduce the bug in my system.
- Create file `mini.vim` with：

```
set nocompatible
set runtimepath^=/home/pree/.local/share/nvim/plugged/coc.nvim
filetype plugin indent on
syntax on
set hidden
```

- Start (neo)vim with command: `vim -u mini.vim`

- Operate vim.

## Screenshots
The following screenshots show that neovim is stuck on ':w' (saving) and node using 100% CPU resources
![s1](https://user-images.githubusercontent.com/63976635/155856305-11e60b5e-1247-4504-9f2a-ceb88f751e62.png)
![s2](https://user-images.githubusercontent.com/63976635/155856345-4ec7efbc-5d0d-4a81-8205-d8584b82e337.png)

**Comment by Preetam-Das** [comment]
Edit: To properly reproduce the bug use a large file(mine was around 1.1 Mb) in a split and try editing another file in another split window which has coc enabled and running.

**Comment by chemzqm** [comment]
@Preetam-Das Can you repoduce it with lastest coc.nvim? Might be fixed by 60321d3f1

**Comment by Preetam-Das** [comment]
Yes. Updated to the latest version but the problem still persists.

```
## versions

vim version: NVIM v0.7.0-dev+1136-g005a7aa16
node version: v17.7.1
coc.nvim version: 0.0.80-a554a31c 2022-03-11 19:04:48 +0800
coc.nvim directory: /home/pree/.local/share/nvim/plugged/coc.nvim
term: alacritty
platform: linux

## Log of coc.nvim

2022-03-11T18:48:25.070 INFO (pid:15312) [plugin] - coc.nvim initialized with node: v17.7.1 after 183ms
2022-03-11T18:48:25.078 INFO (pid:15312) [services] - registered service "clangd"
2022-03-11T18:48:25.079 INFO (pid:15312) [services] - clangd state change: stopped => starting
2022-03-11T18:48:25.094 INFO (pid:15312) [language-client-index] - Language server "clangd" started with 15324
2022-03-11T18:48:25.115 INFO (pid:15312) [services] - clangd state change: starting => running
2022-03-11T18:48:25.123 INFO (pid:15312) [services] - service clangd started
2022-03-11T18:48:28.527 INFO (pid:15312) [attach] - receive notification: showInfo []
2022-03-11T18:50:29.783 INFO (pid:15312) [completion-complete] - 1 results from: around
2022-03-11T18:52:14.243 INFO (pid:15312) [completion-complete] - 1 results from: around
2022-03-11T18:54:12.198 INFO (pid:15312) [completion-complete] - 0 results from: 
2022-03-11T18:59:43.176 INFO (pid:15312) [attach] - receive notification: showInfo []

```
I just disable coc while working with large files.

**Comment by chemzqm** [comment]
It should be blocked on `formatOnSave`, what language server is activated for that large file?

**Comment by Preetam-Das** [comment]
Clangd. Its the same with pyright too. What should be blocked on ```formatOnSave```, I don't get it.

**Comment by fannheyward** [comment]
> Its the same with pyright too.

Do you use `coc-pyright`? `coc-pyright` will run external formatter to format code, which formatter do you use?

**Comment by xiyaowong** [comment]
> Clangd. Its the same with pyright too. What should be blocked on `formatOnSave`, I don't get it.

He doesn't mean should, he means might.

**Comment by fannheyward** [comment]
@Preetam-Das do you still have this issue with pyright `python.formatting.provider` to `none`?

**Comment by Preetam-Das** [comment]
@fannheyward Yes I use ```coc-pyright``` and I set ```"python.formatting.provider": "none"``` in ```coc-settings.json``` but no good.

**Comment by chemzqm** [comment]
Should be fixed by 8ff0de75 `getText()` of `LinesTextDocument` could is slow on big buffer, the solution is use cached string.
